src/mocks.ts

```diff
@@ -47,11 +50,35 @@ export class NavMock {
 }
 
 export class PlatformMock {
-  public ready(): any {
-    return new Promise((resolve: Function) => {
-      resolve();
+  public ready(): Promise<{String}> {
+    return new Promise((resolve) => {
+      resolve('READY');
     });
   }
+
+  public registerBackButtonAction(fn: Function, priority?: number): Function {
+    return (() => true);
+  }
+
+  public hasFocus(ele: HTMLElement): boolean {
+    return true;
+  }
+
+  public doc(): HTMLDocument {
+    return document;
+  }
+
+  public registerListener(ele: any, eventName: string, callback: any): Function {
+    return (() => true);
+  }
+
+  public win(): Window {
+    return window;
+  }
+
+  public raf(callback: any): number {
```

Comment:
What is raf? I think either a comment or more descriptive function name would clear this up.